crates/sui-data-ingestion-core/src/reducer.rs

```diff
@@ -81,4 +81,8 @@ where
     async fn commit(&self, batch: Vec<R>) -> Result<()> {
         self.as_ref().commit(batch).await
     }
+
+    fn should_close_batch(&self, batch: &[R], next_item: Option<&R>) -> bool {
```

Comment:
Was unnecessary/optional in the previous version. Now, it's mandatory. Without this, the default implementation will be called, and the rules for sealing the batch will be ignored

Reply:
wow, very subtle. great catch!